indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -268,6 +272,11 @@ protected void innerMap(
           )
       ).asBytes();
 
+      byte[] serializedInputRow = inputRow instanceof SegmentInputRow ?
```

Comment:
comment added and added a class BatchDeltaIngestionTest that is similar to IndexGeneratorJobTest